File: DonationApp.Core/Entities/Campaign.cs
﻿using System.ComponentModel.DataAnnotations.Schema;
using System.Security.Cryptography;

namespace DonationApp.Core.Entities
{
    public class Campaign : AuditEntity<int>
    {
        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string Location { get; set; } = string.Empty;

        public DateOnly StartDate { get; set; }

        public DateOnly EndDate { get; set; }

        public string OrganizationId { get; set; } = string.Empty;

        [ForeignKey(nameof(OrganizationId))]
        public ApplicationUser? Organization { get; set; }

        public ICollection<Comment>? Comments { get; set; }

        public ICollection<Rating>? Ratings { get; set; }

        public bool IsActivated { get; set; } = true;

        public CampaignAccount CampaignAccount { get; set; } = default!;

        public CampaignLikeCount CampaignLikeCount { get; set; } = default!;

        public ICollection<ApplicationUser>? Subscribers { get; set; }
    }
}

File: DonationApp.UI/src/app/models/campaign.ts
import { CommentResponse } from './comment';

export interface Campaign {
  name: string;
  description: string;
  location: string;
  startDate: Date;
  endDate: Date;
  organizationId: string;
  organizationName: string;
  isActivated: boolean;
  accountNumber: string;
  accountBalance: number;
  id: number;
  isDeleted: boolean;
  comments: CommentResponse[];
  likeCount: number;
}

export interface CreateCampaign {
  name: string;
  description: string;
  location: string;
  startDate: Date;
  endDate: Date;
  organizationId: string;
}
